pkg/client/unversioned/request.go

```diff
@@ -48,6 +48,16 @@ import (
 // are therefore not allowed to set manually.
 var specialParams = sets.NewString("timeout")
 
+// backoff is the underlying backoff struct we use to prevent thundering herd drama, esp when this runs in a controller.
+var backoff = &URLBackoff{Backoff: util.NewBackOff(1*time.Second, 120*time.Second)}
```

Comment:
We do, but there are probably cases where we should wait longer in *some*
cases than what the server recommends.

On Tue, Nov 17, 2015 at 4:57 PM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/client/unversioned/request.go
> <https://github.com/kubernetes/kubernetes/pull/16811#discussion_r45127853>
> :
>
> > @@ -48,6 +48,16 @@ import (
> >  // are therefore not allowed to set manually.
> >  var specialParams = sets.NewString("timeout")
> >
> > +// backoff is the underlying backoff struct we use to prevent thundering herd drama, esp when this runs in a controller.
> > +var backoff = &URLBackoff{Backoff: util.NewBackOff(1*time.Second, 120*time.Second)}
>
> shouldn't we be working off of Retry-After headers in responses? I thought
> parts of that plumbing were already in place
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/16811/files#r45127853>.
>